doc/design/doi-generation-1.md

```diff
@@ -0,0 +1,41 @@
+# Publish process
+
+This process is predicated on a few tasks that are expected to receive valid input and assuming that only certain errors could happen that are out of our control.
+Dandisets have a new attribute added to track their validation status: `VALID`, `INVALID`, and `VALIDATING`.
+Whenever a dandiset is modified (metadata changed or files uploaded/removed), it is set to `VALIDATING` and an asynchronous task is triggered to run metadata validation on it. The task will update the state of the dandiset accordingly when it finishes.
+
+- publish enabled only if a dandiset is `VALID`. The button will be greyed out in the UI and the API endpoint will return 405.
+- on publish trigger:
+   - create the versioned release in the db (this prevents any asset/blob belonging to the release from being deleted)
+   - insert all the relevant metadata (version, doi, url, manifest location/checksum) into the metadata record of the new version 
+  - send this new metadata record to mint the doi (there should be no reason why this should fail unless the external resource fails. if it fails notify admins since this is likely a new bug)
+   - write out to s3 bucket (even on any datacite failure)
+   
+## DOI generation process
+
+## dandi-api
+
+The DOI generation process happens inside a call to `POST /dandisets/{dandiset_id}/draft/publish/`.
+DOI generation is the first step of the publish process, because it is the only step to rely on an external service.
+If the call to the Datacite API fails, there are no files in S3 that need to be cleaned up.
+
+The API will pass the following to the DOI generator:
+* A DOI identifier. This will be of the form `f'{prefix}/{dandiset_id}/{version_id}'`, for example `10.80507/dandi.000001.v0.210402.1835`.
```

Comment:
assuming that we are using "." instead of "/" on purpose?  Should we change the form to something like this?
`f'{prefix}/dandi.{dandiset_id}.v.{version_id}'